SHELL := bash
.ONESHELL:
.SHELLFLAGS := -eu -o pipefail -c
.DELETE_ON_ERROR:
MAKEFLAGS += --warn-undefined-variables
MAKEFLAGS += --no-builtin-rules

##--------------------------------------------##

# SRR number to download reads
SRR=SRR31641679

# Number of reads to download
N=10000

# Downloaded read names
R1=reads/${SRR}_1.fastq
R2=reads/${SRR}_2.fastq

# The reads directory
RDIR=reads

# The accession number of the genome
ACC=GCF_022045245.1

# The name of the genome file
GENOME=Lucila.fa

# The name of the GFF file
GFF=Lucila.gff

##--------------------------------------------##

usage:
	@echo "Usage: make <target>"
	@echo ""
	@echo "Targets:"
	@echo "  reads: Download reads from SRR number"
	@echo "  download: Download the genome and GFF files"
	@echo "  index: Index the genome file"
	@echo "  align: Align the reads to the genome"
	@echo "  all: Download reads, download genome, index genome, and align reads"
	@echo "  clean: Remove all files generated by this Makefile"

reads:
	mkdir -p ${RDIR}

	# Download the SRR data
	fastq-dump --split-files -X ${N} -O ${RDIR} ${SRR}

download:  
	# Download fasta file from NCBI
	datasets download genome accession ${ACC} --include genome,gff3

	# Unzip the data (Overwrite files if they already exist)
	unzip -o ncbi_dataset.zip

	# Make a link to a simpler genome name
	ln -s ncbi_dataset/data/GCF_022045245.1/GCF_022045245.1_ASM2204524v1_genomic.fna Lucila.fa

	# Make a link to a simpler GFF name
	ln -s ncbi_dataset/data/GCF_022045245.1/genomic.gff Lucila.gff

index: 
	# Generate hiseq2 index
	hisat2-build ${GENOME} ${GENOME}-index

align: 
	# Make the bam directory
	mkdir -p bam

	# Align the reads to the genome using hisat2
	hisat2 -x ${GENOME}-index -1 ${R1} -2 ${R2} -S bam/${SRR}_aligned_reads.sam

	# Convert the sam file to a bam file
	samtools view -b -o bam/${SRR}_aligned_reads.bam bam/${SRR}_aligned_reads.sam

	# Sort the bam file
	samtools sort -o bam/${SRR}_aligned_reads.sorted.bam bam/${SRR}_aligned_reads.bam

	# Index the bam file
	samtools index bam/${SRR}_aligned_reads.sorted.bam

	# Create a bigwig coverage file
	bamCoverage -b bam/${SRR}_aligned_reads.sorted.bam -o bam/${SRR}_aligned_reads.bw

	# Index the bigwig file
	bigWigIndex bam/${SRR}_aligned_reads.bw

all: reads download index align

clean:
	rm -rf ${RDIR} ${GENOME} ${GFF} bam Lucila.fa-index.1.ht2 Lucila.fa-index.2.ht2 Lucila.fa-index.3.ht2 Lucila.fa-index.4.ht2 Lucila.fa-index.5.ht2 Lucila.fa-index.6.ht2 Lucila.fa-index.7.ht2 Lucila.fa-index.8.ht2 md5sum.txt ncbi_dataset.zip ncbi_dataset README.md 

.PHONY: usage all clean reads download index align





